metrics.py

```diff
@@ -0,0 +1,82 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+import networkx as nx
+
+from utils import *
+
+## TODO - Metrics needs to implement a diff() function
+class Metric(ABC):
+    def __init__(self, metric_name, requires_graph=False):
```

Comment:
Maybe `Metric` should have a field of `support_directed` which would specify which kind of graphs it supports, and this will be validated against the `is_directed` field of `MetricCollection` in its constructor. This will allow to spare validity tests at the level of each `Metric`, by raising an error earlier, and will keep the API such that `calculate` gets only an input matrix/graph.
In this way, for example, `NumberOfEdges` will be split into 2: `NumberOfEdgesDirected` and  `NumberOfEdgesUnDirected`

Reply:
Added as a TODO. lets discuss this